tests/move_architecture/test_architecture.py

```diff
@@ -0,0 +1,107 @@
+    arch = QuantumArchitecture(**move_architecture_json)
+    check_arch(arch.quantum_architecture)
+
+
+def check_arch(arch: QuantumArchitectureSpecification):
+    """Verifies architecture is as expected"""
+    assert arch.name == 'Custom arch'
+    assert arch.qubits == ['COMP_R', 'QB1', 'QB2', 'QB3', 'QB4', 'QB5', 'QB6']
+    assert arch.qubit_connectivity == [
+        ['QB1', 'COMP_R'],
+        ['QB2', 'COMP_R'],
+        ['QB3', 'COMP_R'],
+        ['QB4', 'COMP_R'],
+        ['QB5', 'COMP_R'],
+        ['QB6', 'COMP_R'],
+    ]
+    assert arch.operations['prx'] == [['QB1'], ['QB2'], ['QB3'], ['QB4'], ['QB5'], ['QB6']]
+    assert arch.operations['cz'] == [
+        ['QB1', 'COMP_R'],
+        ['QB2', 'COMP_R'],
+        ['QB3', 'COMP_R'],
+        ['QB4', 'COMP_R'],
+        ['QB5', 'COMP_R'],
+    ]
+    assert arch.operations['move'] == [['QB6', 'COMP_R']]
+
+
+def test_backend_configuration_new(new_architecture):
+    """Check that the extended architecture is configured correctly to the Qiskit backend"""
+    assert new_architecture is not None
+    backend, _client = get_mocked_backend(new_architecture)
+    assert backend.target.physical_qubits == [0, 1, 2, 3, 4, 5, 6]
+    assert set(backend.target.operation_names) == {'r', 'id', 'cz', 'measure', 'move'}
+    assert [f'{o.name}:{o.num_qubits}' for o in backend.target.operations] == [
+        'r:1',
+        'id:1',
+        'cz:2',
+        'measure:1',
+        'move:2',
+    ]
+    check_instruction(backend.instructions, 'r', [(1,), (2,), (3,), (4,), (5,), (6,)])
+    check_instruction(backend.instructions, 'measure', [(1,), (2,), (3,), (4,), (5,), (6,)])
+    check_instruction(backend.instructions, 'id', [(0,), (1,), (2,), (3,), (4,), (5,), (6,)])
+    check_instruction(
+        backend.instructions, 'cz', [(1, 0), (0, 1), (2, 0), (0, 2), (3, 0), (0, 3), (4, 0), (0, 4), (5, 0), (0, 5)]
+    )
+    check_instruction(backend.instructions, 'move', [(6, 0)])
+
+
+def test_backend_configuration_adonis(adonis_architecture):
+    """ "Check that the Qiskit backend configuration still works properly for Adonis architecture"""
+    assert adonis_architecture is not None
+    backend, _client = get_mocked_backend(adonis_architecture)
+    assert backend.target.physical_qubits == [0, 1, 2, 3, 4]
+    assert set(backend.target.operation_names) == {'r', 'id', 'cz', 'measure'}
+    assert [f'{o.name}:{o.num_qubits}' for o in backend.target.operations] == [
+        'r:1',
+        'id:1',
+        'cz:2',
+        'measure:1',
+    ]
+    check_instruction(backend.instructions, 'r', [(0,), (1,), (2,), (3,), (4,)])
+    check_instruction(backend.instructions, 'measure', [(0,), (1,), (2,), (3,), (4,)])
+    check_instruction(backend.instructions, 'id', [(0,), (1,), (2,), (3,), (4,)])
+    check_instruction(backend.instructions, 'cz', [(0, 2), (2, 0), (1, 2), (2, 1), (3, 2), (2, 3), (4, 2), (2, 4)])
+
+
+def check_instruction(
```

Comment:
This can be part of iqm-client validation logic.

Reply:
I've now added validation logic to IQM client that checks that operations are targeting correct qubits (loci are defined in the architecture specification), but this check is for testing that the qiskit backend is configured correctly, so that qiskit's algorithms work properly. The Instruction here is qiskit's Instruction. So I need to have these checks here as well.